packages/block-editor/src/hooks/block-rename-ui.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * WordPress dependencies
+ */
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { addFilter } from '@wordpress/hooks';
+import { __ } from '@wordpress/i18n';
+import { getBlockSupport } from '@wordpress/blocks';
+import {
+	MenuItem,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	Button,
+	TextControl,
+	Modal,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	BlockSettingsMenuControls,
+	useBlockDisplayInformation,
+} from '../components';
+
+const emptyString = ( testString ) => testString?.trim()?.length === 0;
+
+function RenameModal( { blockName, originalBlockName, onClose, onSave } ) {
+	const [ editedBlockName, setEditedBlockName ] = useState( blockName );
+
+	const nameHasChanged = editedBlockName !== blockName;
+
+	const isNameValid = nameHasChanged && ! emptyString( editedBlockName );
+
+	return (
+		<Modal
+			title={ __( 'Rename block' ) }
+			onRequestClose={ onClose }
+			overlayClassName="block-editor-block-rename-modal"
+		>
+			<p>{ __( 'Choose a custom name for this block.' ) }</p>
+			<form
+				onSubmit={ ( e ) => {
+					e.preventDefault();
+
+					if ( ! isNameValid ) {
+						return;
+					}
+
+					onSave( editedBlockName );
+
+					// Immediate close avoids ability to hit save multiple times.
+					onClose();
+				} }
+			>
+				<VStack spacing="3">
+					<TextControl
+						__nextHasNoMarginBottom
+						value={ editedBlockName }
+						label={ __( 'Block name' ) }
+						hideLabelFromVision={ true }
```

Comment:
This might be a problem since placeholder text holds the current block name and will disappear when empty or typed into. The modal title may be enough if it is visible visually.

Reply:
Sorry Alex I didn't follow this feedback. I'd be grateful if you could explain a little more as to what you expect here? Much appeciated.